rpcs3/Emu/Cell/lv2/sys_tty.cpp

```diff
@@ -1,17 +1,60 @@
 #include "stdafx.h"
 #include "Emu/system_config.h"
+#include "Emu/IdManager.h"
+#include "Emu/System.h"
 
 #include "sys_tty.h"
 
 #include <deque>
-#include <mutex>
+#include <shared_mutex>
 
 LOG_CHANNEL(sys_tty);
 
-extern fs::file g_tty;
-extern atomic_t<s64> g_tty_size;
 extern std::array<std::deque<std::string>, 16> g_tty_input;
-extern std::mutex g_tty_mutex;
+
+tty_t::tty_t()
+{
+	const auto tty_path = fs::get_cache_dir() + "TTY.log";
+	fd.open(tty_path, fs::write + fs::read + fs::create);
+
+	if (!fd)
+	{
+		sys_tty.error("Failed to create TTY log: %s (%s)", tty_path, fs::g_tls_error);
+	}
+	else
+	{
+		// Either title ID or name of the file
+		const std::string name = Emu.GetTitleID().empty() ? Emu.argv[0].substr(Emu.argv[0].find_last_of(fs::delim) + 1) : Emu.GetTitleID();
+
+		char last = '\n';
+
+		start_pos = fd.seek(0, fs::seek_end);
+
+		if (start_pos)
+		{
+			// Get last character
+			fd.seek(start_pos - 1);
+			fd.read(last);
+		}
+
+		start_pos = fd.seek(0, fs::seek_end);	
+		fd.write(fmt::format("%sTTY of %s:\n", last != '\n' ? "\n" : "", name));
```

Comment:
I think the name declaration should be closer to its usage. I was wondering where it's used until I finally got to the end of the scope.

I'd assume that this would look better visually (not tested):
```suggestion
		fd.write(fmt::format("%sTTY of %s:\n", last != '\n' ? "\n\n" : (start_pos ? "\n" : ""), name));
```
Actually, why not make it crystal clear what happens? Something like:
"RPCS3 is booting %s. TTY:"
```suggestion
		fd.write(fmt::format("%sRPCS3 is booting %s. Printing TTY:\n", last != '\n' ? "\n\n" : (start_pos ? "\n" : ""), name));
```

Reply:
Maybe don't add the boot part after all if you print it to the regular log as well. might confuse users.

Reply:
I want as little \n as possible, like if previous text from previous application already had it theres no reason to readd it.